dialer.go

```diff
@@ -275,9 +277,18 @@ func newInstrumentedConn(conn net.Conn, closeFunc func()) *instrumentedConn {
 // is closed.
 type instrumentedConn struct {
 	net.Conn
+	// rawConn is the underlying net.Conn without TLS
+	rawConn   net.Conn
 	closeFunc func()
 }
 
+// SyscallConn supports a connection check in the MySQL driver by delegating to
+// the underlying non-TLS net.Conn.
+func (i *instrumentedConn) SyscallConn() (syscall.RawConn, error) {
+	sconn := i.rawConn.(syscall.Conn)
```

Comment:
will this break on windows?

Reply:
Good question. I'll a unit test to make sure we have support in either case.

Reply:
Looks like the MySQL driver doesn't have support for Windows here: https://github.com/go-sql-driver/mysql/blob/ad9fa14acdcf7d0533e7fbe58728f3d216213ade/conncheck_dummy.go#L9. What would your expectation be?

Reply:
I've updated this so it won't panic if the type assertion fails. Separately, this builds on Windows with the unit test, so I think we're good.

Reply:
I'll take another look at this and see what we can do. Thanks for all your help on this.

Reply:
It looks like the platforms that don't support the call won't use connCheck: https://github.com/go-sql-driver/mysql/blob/ad9fa14acdcf7d0533e7fbe58728f3d216213ade/conncheck_dummy.go#L16

So it seems like it's probably fine to return an error, since if it is being used something probably wrong?

Reply:
That’s my understanding as well. @Gentoli what do you think?

Reply:
Sorry I did not realize `// +build` is built-in to golang. Both with or without the check on the cast should work.